Lib/typing.py

```diff
@@ -297,8 +305,27 @@ def _check_generic(cls, parameters, elen):
         raise TypeError(f"{cls} is not a generic class")
     alen = len(parameters)
     if alen != elen:
-        raise TypeError(f"Too {'many' if alen > elen else 'few'} arguments for {cls};"
-                        f" actual {alen}, expected {elen}")
+        expect_val = elen
+        if hasattr(cls, "__parameters__"):
+            parameters = [p for p in cls.__parameters__ if get_origin(p) is not Unpack]
+
+            # deal with TypeVarLike defaults
+            # required TypeVarLikes cannot appear after a defaulted one.
+            if alen < elen:
+                # since we validate TypeVarLike default in _collect_type_vars
+                # or _collect_parameters we can safely check parameters[alen]
+                if getattr(parameters[alen], '__default__', None) is not None:
```

Comment:
```suggestion
                if hasattr(parameters[alen], '__default__'):
```

Reply:
Oh, I suppose this is deliberately written the way it is currently so that TypeVarLikes with defaults can be differentiated from those without them, since a `__default__` attribute set to `None` has special meaning in this context?

Reply:
I copied this from typing-extensions without thinking too hard; I wonder if it's fine to do:
```suggestion
                if parameters[alen].__default__ is not None:
```
Because in CPython we can assume the TypeVars also come from us and therefore must have a `__default__` attribute.

Reply:
Applied my suggestion as it passes tests.